### PR TITLE
Upgrade controller-runtime version to v0.7.0

### PR DESCRIPTION
**Pull request by andreyod**
controller-runtime v0.7.0 introduces various breaking ang high impact changes.

For full list of changes see:
https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.7.0

Relevant changes:
* Use client.Object and client.ObjectList
* Add a context to Reconciler interface
* client.ObjectKeyFromObject now uses client.Object
* Upgrade v1beta1 admission types to v1
* Add predicate for annotations change on update event
* Simplify usage of EnqueueRequestsFromMapFunc

Kubevirt related change:
Stop using kubevirt.io/kubevirt/pkg/virt-config package as it eventually uses incompatible k8s package.

**Release note**:

```release-note
Upgrade controller-runtime to v0.7.0
```

### CODE REVIEW

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -182,7 +183,7 @@ type ReconcileHyperConverged struct {
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *ReconcileHyperConverged) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```
Comment (nunnatsa) [inline pkg/controller/hyperconverged/hyperconverged_controller.go:186]: OK, this is new... But if we finally receiving the context from the SDK, we need to stop use ours. Please use it in `common.NewHcoRequest` (will require to change its signature, to add `ctx context.Context` as its first parameter).

**Review by nunnatsa — CHANGES_REQUESTED** [review]
Just one comment about the new context parameter in `Reconcile`

File: pkg/controller/hyperconverged/hyperconverged_controller.go (hunk 1 above)
Comment (andreyod) [reply pkg/controller/hyperconverged/hyperconverged_controller.go:186]: Done

**Pull request closed without merging**